components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/utils/APIUtilTest.java

```diff
@@ -1946,7 +2038,429 @@ public void testIsRoleExistForUser() throws Exception {
+        PowerMockito.doReturn(tierMap).when(APIUtil.class, "getTiersFromPolicies", PolicyConstants.POLICY_LEVEL_SUB, tenantID);
+
+        Map<String, Tier> appTierMap = APIUtil.getTiers(tierType, tenantDomain);
+        Assert.assertEquals(tierMap, appTierMap);
+    }
+
+    @Test
+    public void testGetTiersWhenTierTypeIsResource() throws Exception {
+
+        System.setProperty(CARBON_HOME, "");
+        int tierType = APIConstants.TIER_RESOURCE_TYPE;
+        int tenantID = 1;
+        String tenantDomain = "Wso2.com";
+        Tier tier1 = Mockito.mock(Tier.class);
+        Map<String, Tier> tierMap = new TreeMap<String, Tier>();
+        tierMap.put("UNLIMITED", tier1);
+
+        PowerMockito.mockStatic(PrivilegedCarbonContext.class);
+        PowerMockito.doNothing().when(PrivilegedCarbonContext.class, "startTenantFlow");
+
+        PrivilegedCarbonContext privilegedCarbonContext = Mockito.mock(PrivilegedCarbonContext.class);
+        Mockito.when(PrivilegedCarbonContext.getThreadLocalCarbonContext()).thenReturn(privilegedCarbonContext);
+
+        doNothing().when(privilegedCarbonContext).setTenantDomain(tenantDomain, true);
+
+        Mockito.when(privilegedCarbonContext.getTenantId()).thenReturn(tenantID);
+
+        PowerMockito.spy(APIUtil.class);
+        PowerMockito.doReturn(tierMap).when(APIUtil.class, "getTiersFromPolicies", PolicyConstants.POLICY_LEVEL_API, tenantID);
+
+        Map<String, Tier> appTierMap = APIUtil.getTiers(tierType, tenantDomain);
+
+        Assert.assertEquals(tierMap, appTierMap);
+    }
+
+    @Test
+    public void testGetTiersWhenTierTypeIsApplication() throws Exception {
+
+        System.setProperty(CARBON_HOME, "");
+        int tierType = APIConstants.TIER_APPLICATION_TYPE;
+        int tenantID = 1;
+        String tenantDomain = "Wso2.com";
+        Tier tier1 = Mockito.mock(Tier.class);
+        Map<String, Tier> tierMap = new TreeMap<String, Tier>();
+        tierMap.put("UNLIMITED", tier1);
+
+        PowerMockito.mockStatic(PrivilegedCarbonContext.class);
+        PowerMockito.doNothing().when(PrivilegedCarbonContext.class, "startTenantFlow");
+
+        PrivilegedCarbonContext privilegedCarbonContext = Mockito.mock(PrivilegedCarbonContext.class);
+        Mockito.when(PrivilegedCarbonContext.getThreadLocalCarbonContext()).thenReturn(privilegedCarbonContext);
+
+        doNothing().when(privilegedCarbonContext).setTenantDomain(tenantDomain, true);
+
+        Mockito.when(privilegedCarbonContext.getTenantId()).thenReturn(tenantID);
+
+        PowerMockito.spy(APIUtil.class);
+        PowerMockito.doReturn(tierMap).when(APIUtil.class, "getTiersFromPolicies", PolicyConstants.POLICY_LEVEL_APP, tenantID);
+
+        Map<String, Tier> appTierMap = APIUtil.getTiers(tierType, tenantDomain);
+
+        Assert.assertEquals(tierMap, appTierMap);
+    }
+
+    @Test
+    public void testValidateAPICategoriesWithValidCategories() throws Exception {
+
+        String tenantDomain = "Wso2.com";
```

Comment:
Shall we define `tenantDomain` as class variable since it is used in multiple places?

Reply:
this is done